python/whylogs/core/validators/condition_validator.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+from itertools import chain
+from typing import Any, Callable, Dict, List
+
+from whylogs.core.metrics.condition_count_metric import Condition
```

Comment:
Why use Condition? The name sounds fitting but it doesn't appear to simplify the code, seems like you just need a function that returns bool?

Reply:
Having predefined conditions feel like the right way, but I don't have a good argument for that. Maybe to simplify serialization in the future?
@richard-rogers , any thoughts on this?

Reply:
`Condition` should be split-up/moved out of `condition_count_metric.py` since the goal is to share the predicate infrastructure between the condition count metric and validators. I've made a start at this in the binary condition PR #781

Reply:
I think @jamie256's question was about the need to create conditions with Relations, when a lambda expression could cover all of our cases? or maybe any function that returns a bool.

Reply:
Originally, `Condition` carried a little metadata in addition to the `Callable` implementing the predicate.  We should definitely refactor to share between validators and metrics. #781 is a start, but could probably use more refactoring still.

Reply:
If you are adding more functionality to Condition then it seems better justified. At present the benefits of `Condition `weren't clear over using a lambda or predicate function and requiring a class for `Condition` increases complexity in how users have to construct validators so its seemed to detract from usability.